### PR TITLE
Updated architecture description

### PR DESCRIPTION
**Pull request by hannestschofenig**

### CODE REVIEW

File: draft-bft-rats-kat.md [hunk 1]
```diff
@@ -119,22 +119,38 @@ in {{Section 8 of -cbor}} and {{Appendix G of -cddl}}.
...
- | +-------------+  +-------------+ |
- +----------------------------------+
-       ^
+ +----------------------------------+ .
+ | Attester                         | .
+ |                                  | .
+ | +-------------+  +-------------+ | .
+ | | Key         |  | Platform    | | .
+ | | Attestation |  | Attestation | | .
+ | | Service(KAS)|  | Service     | | .
```
Comment (thomas-fossati) [inline draft-bft-rats-kat.md]:
```suggestion
 | | Service     |  | Service     | | .
```

File: draft-bft-rats-kat.md [hunk 2]
```diff
@@ -119,22 +119,38 @@ in {{Section 8 of -cbor}} and {{Appendix G of -cddl}}.
...
-In an architecture for key attestation there are three or more parties communicating with
-each other. From a key attestation point of view, there are the Key Attestation Service,
-the Presenter and the Recipient. Additional parties are added when considering attestation-
-related entities.
+Key attestation is an extension to the attestation functionality described in {{-rats-arch}}. 
+We describe this conceptually by splitting the internals of the attester into a two parts, 
+platform attestation and key attestation. This is shown in {{fig-arch}}. These are logical
+roles and implementations may combine them into a single physical entity. 
+
+Security-sensitive functionality, like the attestation functionality, has to be placed into
```
Comment (thomas-fossati) [inline draft-bft-rats-kat.md]:
```suggestion
Security-sensitive functionality, like attestation, has to be placed into
```

File: draft-bft-rats-kat.md [hunk 3]
```diff
@@ -119,22 +119,38 @@ in {{Section 8 of -cbor}} and {{Appendix G of -cddl}}.
...
+the trusted computing base. Since the trusted computing base itself may support different 
+isolation layers, the design allows platform attestation to be separated from key attestation
+whereby platform attestation requires is more privileged than the key attestation code.
```
Comment (thomas-fossati) [inline draft-bft-rats-kat.md]:
```suggestion
whereby platform attestation requires more privilege than the key attestation code.
```

File: draft-bft-rats-kat.md [hunk 4]
```diff
@@ -119,22 +119,38 @@ in {{Section 8 of -cbor}} and {{Appendix G of -cddl}}.
...
+Cryptographic services, used by key attestation and by platform attestation, are separated
+although not shown in the figure.
+
+The protocol used for communication between the Presenter and the Recipient is referred as 
+usage protocol. The usage protocol is outside the scope of this specification, needs to
```
Comment (thomas-fossati) [inline draft-bft-rats-kat.md]:
```suggestion
usage protocol. The usage protocol, which is outside the scope of this specification, needs to
```

File: draft-bft-rats-kat.md [hunk 5]
```diff
@@ -145,49 +161,63 @@ related entities.
 ~~~
 {: #fig-arch title="Architecture"}
 
-The Presenter generates the IK, which contains a public key (pkT) and a private key (skT), for example
-using the following API call:
+The Presenter triggers the generation of the IK. The IK
+consists of a public key (pkT) and a private key (skT).
```
Comment (thomas-fossati) [inline draft-bft-rats-kat.md:165]: just a curiosity, what does `T` means in `[ps]kT` ?

File: draft-bft-rats-kat.md [hunk 6]
```diff
@@ -145,49 +161,63 @@ related entities.
...
+The Recipient of the CAB and the proof-of-possession data (such
+as a digital signature) first extracts the PAT and the KAT. The
+PAT and the KAT may need to be conveyed to a Verifier. If the 
+PAT is in the form of attestation results the checks can be 
+performed locally at the Recipient, whereby the following checks
+are made:
+
+- The signature protecting the PAT MUST pass verification when 
+using available trust anchor(s).
+- The PAT MUST be checked for replays, which can be checked by
```
Comment (thomas-fossati) [inline draft-bft-rats-kat.md]: IIUC, the bundling method we describe reuses the PAT across multiple KATs, so the session freshness for the "usage protocol" is to be checked in the KAT only.

This step should be "PAT and KAT chaining is checked"

**Review by thomas-fossati — COMMENTED** [review]
great stuff.  my usual set of assorted editorial comments inlined.

**Comment by thomas-fossati** [comment]
The CI pipeline is broken:

```
draft-bft-rats-kat.md contains trailing whitespace
*** Run 'make fix-lint' to automatically fix some errors
```
and
```
draft-bft-rats-kat.md does not contain its own name (draft-bft-rats-kat-latest)
*** Correct the name of drafts in docname or similar fields
```

File: draft-bft-rats-kat.md [hunk 7]
```diff
@@ -3,9 +3,8 @@ v: 3
 
 title: "Key Attestation Token"
```
Comment (yogeshbdeshpande) [inline draft-bft-rats-kat.md:4]: The Name Key Attestation itself is mis-leading

File: draft-bft-rats-kat.md [hunk 8]
```diff
@@ -119,22 +119,38 @@ in {{Section 8 of -cbor}} and {{Appendix G of -cddl}}.
 
 # Architecture
 
-In an architecture for key attestation there are three or more parties communicating with
-each other. From a key attestation point of view, there are the Key Attestation Service,
-the Presenter and the Recipient. Additional parties are added when considering attestation-
-related entities.
+Key attestation is an extension to the attestation functionality described in {{-rats-arch}}.
```
Comment (yogeshbdeshpande) [inline draft-bft-rats-kat.md:122]: It should be better termed as "Augmented Key Attestation" or "Enhanced Key Attestation"

Comment (thomas-fossati) [reply draft-bft-rats-kat.md:122]: > It should be better termed as "Augmented Key Attestation" or "Enhanced Key Attestation"

not sure, we are not enhancing or augmenting key attestation

File: draft-bft-rats-kat.md (hunk 7 above)
Comment (thomas-fossati) [reply draft-bft-rats-kat.md:4]: > The Name Key Attestation itself is mis-leading

in what sense?

Comment (yogeshbdeshpande) [reply draft-bft-rats-kat.md:4]: We are not documenting Key Attestation here!

Comment (thomas-fossati) [reply draft-bft-rats-kat.md:4]: 😕

File: draft-bft-rats-kat.md (hunk 5 above)
Comment (hannestschofenig) [reply draft-bft-rats-kat.md:165]: The text was initially in the TLS attestation draft and there it was called TLS Identity Key. The T was related to the TLS part. Now, we are only talking about the Identity Key anymore and hence the abbreviation is less useful....

**Pull request merged**